File: benchmarks/cross/cases/data/hashmap_loop.go
package main

import (
	"fmt"
	"os"
	"strconv"
)

const valueModulus = int64(1000003)

func requestedWork() int64 {
	if len(os.Args) < 2 {
		return 1
	}
	work, err := strconv.ParseInt(os.Args[1], 10, 64)
	if err != nil {
		return 1
	}
	if work < 0 {
		return 0
	}
	return work
}

func main() {
	work := requestedWork()
	entries := make(map[int64]int64, int(work))
	for i := int64(0); i < work; i++ {
		entries[i] = (17*i + 3) % valueModulus
	}

	var checksum int64
	for i := work - 1; i >= 0; i-- {
		if value, found := entries[i]; found {
			checksum += value
		}
	}
	fmt.Println(checksum)
}

File: benchmarks/cross/cases/data/hashmap_loop.py
import sys


VALUE_MODULUS = 1_000_003


def requested_work() -> int:
    if len(sys.argv) < 2:
        return 1
    try:
        return max(0, int(sys.argv[1]))
    except ValueError:
        return 1


def main() -> None:
    work = requested_work()
    entries: dict[int, int] = {}
    for key in range(work):
        entries[key] = (17 * key + 3) % VALUE_MODULUS

    checksum = 0
    for key in range(work - 1, -1, -1):
        value = entries.get(key)
        if value is not None:
            checksum += value
    print(checksum)


if __name__ == "__main__":
    main()
